app/server/lib/OIDCConfig.ts

```diff
@@ -187,14 +219,7 @@ export class OIDCConfig {
       ...(agents.trusted !== undefined ? {agent: agents.trusted} : {}),
       ...(httpTimeout !== undefined ? {timeout: httpTimeout} : {}),
     });
-    await this._initClient({ issuerUrl, clientId, clientSecret, extraMetadata });
-
-    if (this._client.issuer.metadata.end_session_endpoint === undefined &&
-      !this._endSessionEndpoint && !this._skipEndSessionEndpoint) {
-      throw new Error('The Identity provider does not propose end_session_endpoint. ' +
-        'If that is expected, please set GRIST_OIDC_IDP_SKIP_END_SESSION_ENDPOINT=true ' +
-        'or provide an alternative logout URL in GRIST_OIDC_IDP_END_SESSION_ENDPOINT');
-    }
```

Comment:
Is `initOIDC` still needs to remain async after your change here?

Making it synchronous could make some parts of the code simpler if you want to take a look (we may remove the `build()` method and even move all the code of the `init()` method in the constructor).